classes/publication/PKPPublication.php

```diff
@@ -70,72 +70,100 @@ public function getLocalizedData($key, $preferredLocale = null, &$selectedLocale
     /**
      * Combine the localized title, prefix and subtitle
      *
-     * @param string $preferredLocale Override the publication's default
-     *  locale and return the title in a specified locale.
-     *
+     * @param  string $preferredLocale  Override the publication's default locale and return the title in a specified locale.
+     * @param  string $format           Define the return data format as text or html
+     * 
      * @return string
      */
-    public function getLocalizedFullTitle($preferredLocale = null)
+    public function getLocalizedFullTitle($preferredLocale = null, string $format = 'text')
     {
-        $fullTitle = $this->getLocalizedTitle($preferredLocale);
-        $subtitle = $this->getLocalizedData('subtitle', $preferredLocale);
+        $fullTitle = $this->getLocalizedTitle($preferredLocale, $format);
+        $subtitle = $this->getLocalizedSubTitle($preferredLocale, $format);
+
         if ($subtitle) {
             return PKPString::concatTitleFields([$fullTitle, $subtitle]);
         }
+
         return $fullTitle;
     }
 
     /**
      * Return the combined prefix, title and subtitle for all locales
      *
+     * @param  string $format Define the return data format as text or html
      * @return array
      */
-    public function getFullTitles()
+    public function getFullTitles(string $format = 'text')
     {
         $allTitles = (array) $this->getData('title');
         $return = [];
         foreach ($allTitles as $locale => $title) {
             if (!$title) {
                 continue;
             }
-            $return[$locale] = $this->getLocalizedFullTitle($locale);
+            $return[$locale] = $this->getLocalizedFullTitle($locale, $format);
         }
         return $return;
     }
 
     /**
      * Combine the localized title and prefix
      *
-     * @param string $preferredLocale Override the publication's default
-     *  locale and return the title in a specified locale.
-     *
+     * @param  string $preferredLocale  Override the publication's default locale and return the title in a specified locale.
+     * @param  string $format           Define the return data format as text or html
+     * 
      * @return string
      */
-    public function getLocalizedTitle($preferredLocale = null)
+    public function getLocalizedTitle($preferredLocale = null, string $format = 'text')
     {
         $usedLocale = null;
         $title = $this->getLocalizedData('title', $preferredLocale, $usedLocale);
         $prefix = $this->getData('prefix', $usedLocale);
+
+        if (strtolower($format) === 'text') {
+            $title = strip_tags($title);
+        }
+        
         if ($prefix) {
```

Comment:
If `$format` is `html`, and `$prefix` contains any HTML-like content, it'll be passed through unescaped. The logic should be something like (without respect to my other note about constants):
```php
$usedLocale = null;
$title = $this->getLocalizedData('title', $preferredLocale, $usedLocale);
$prefix = $this->getData('prefix', $usedLocale);

switch ($format) {
    case 'html':
        // Title is already in HTML, prefix is in text. Convert prefix.
        if ($prefix) $prefix = htmlspecialchars($prefix);
        break;
    case 'text':
        // Title is in HTML, prefix is already in text. Convert title.
        $title = strip_tags($title);
        break;
    default: throw new Exception('Invalid format!');
}

if ($prefix) {
    return $prefix . ' ' . $title;
}
return $title;
```

Reply:
updated as per suggested . One question , as user allow to enter any text in for `prefix`, shouldn't we run it through `PKPString::stripUnsafeHtml` also ? I see that it's not done previously but I think we should just to be safe .

Reply:
Users can enter anything in `prefix`, but it should be treated like plaintext, not HTML. Thus calling `htmlspecialchars` on it ensures that `<`, `>`, and `&` are presented as characters, rather than interpreted as HTML entities. No need to call `stripUnsafeHtml` on top of proper escaping as there's already no room for HTML to be injected.

Reply:
The HTML entities (e.g. `&amp;`) are being left on the `text` format. I've flagged this while reviewing a [PR](https://github.com/pkp/citationStyleLanguage/pull/122) from @bozana, perhaps she will fix there, so I'm just leaving a note here.